NEWS.md

```diff
@@ -32,6 +32,12 @@
 * in `leftjoin`, `rightjoin`, and `outerjoin` the `indicator` keyword argument
   is deprecated in favor of `source` keyword argument; `indicator` will be removed
   in 2.0 release ([2649](https://github.com/JuliaData/DataFrames.jl/pull/2649))
+* legacy `convert` methods for `AbstractDataFrame`, `DataFrameRow` and `GroupKey`
+  are now deprecated as they are not required now; the only conversions that are
+  retained are `convert(::Type{NamedTuple}, dfr::DataFrameRow)`,
+  `convert(::Type{NamedTuple}, key::GroupKey)`, and
+  `convert(::Type{DataFrame}, sdf::SubDataFrame)`; the deprecated methods will be
+  removed in 2.0 release
```

Comment:
Removing them in 2.0 is a conservative approach. Personally I would be OK to remove them earlier (especially because I consider these methods legacy as commented above), but this would break semver, so for now I was conservative.

@oxinabox - I view you in this area as an expert and a representative of user's perspective. What do you think we should do?

In general I see three things we can do:
* make a 1.0 release that is non breaking, make a 2.0 release removing things that are deprecated (this is what we promised)
* make a hard removal of deprecated functionality in some minor release - e.g. 1.1 - this would break semver, but we could already clearly signal now when the removal would happen
* make a 0.23 release that would be considered a beta of 1.0 with depredations and then relatively quickly a 1.0 release with removed deprecations but no new functionality (in a style like Julia 0.7 and Julia 1.0 were released)

I feel that the first approach (1.0 release, and then 2.0 release removing the deprecated methods) is the right one, but it would mean that we would probably stick with these deprecations for another 1 or 2 years.

Reply:
The invalidations it causes are not terrible. It's only like 150.
Conversely people have been waiting for so long for a 1.0 release, and we keep fobbing them off with a one last release before.

Option 2 of dropping in 1.1 is a no-go. it breaks semver.

Good options are:
Option 1: Make a nonbreaking 1.0 release. Remove these in a few years in 2. Probably fine since it is only 150 invalidations

Option 3. One last 0.23. I wouldn't mark it as a beta of 1.0 (that is basically what 0.22 was pitched as). I would describe it as a "oops forgot to remove these before. Most people should update straight from 0.22 to 1.0", and i would release 0.23 and 1.0 on the same day.

I would also like to offer option 4:
skip deprecation and just delete them and release version 1.
These are prettty obsure methods i think, and since people will be updating to use 1.0 basically at the same time as they drop 0.22, a deprecation is not that useful really.
Probably this is less good than option 3.

and option 5: 
deprecate them in release 0.22.6.
DataStructures.jl does this a lot, deprecations in a final nobreaking, before a breaking release.
This is acceptable since it is nonbreaking.
and it fits the workflow of _"try and update to new. Run tests, did anything error? If so downgrade and rerun with deprecations turned on"_, since when you downgrade you will get the latest version of the 0.22
Its basically a bug in 0.22 anyway, since we thought that 0.22 removed all the legacy cruft but it missed some.

I think i like option 5 best

Reply:
Thank you for these thoughts. I think option 5 makes sense and, if @nalimilan agreed to this approach, I would bakcport this PR to 0.22 branch after merging it and then remove the deprecations before the 1.0 release.

This makes sense especially as I assume that these methods are not really used in practice at all (of course I might be wrong 😄).

Reply:
Yes option 5 sounds the most correct WRT semver. It sounds better than carrying these rarely used methods until 2.0.